File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> participant_count;

    // 참가자 목록을 해시맵에 추가
    for (const string& p : participant) {
        participant_count[p]++;
    }

    // 완주자 목록을 차감
    for (const string& c : completion) {
        participant_count[c]--;
        if (participant_count[c] == 0) {
            participant_count.erase(c);
        }
    }

    // 남아 있는 참가자 반환
    return participant_count.begin()->first;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> completionMap = new HashMap<>();
        
        for (String runner : completion) {
            completionMap.merge(runner, 1, Integer::sum);
        }
        
        for (String runner : participant) {
            int remaining = completionMap.merge(runner, -1, Integer::sum);
            if (remaining < 0) {
                return runner;
            }
        }
        
        return null;
    }
}